core/src/main/scala/org/ensime/util/DiffUtil.scala

```diff
@@ -23,8 +25,39 @@ object DiffUtil {
     if (file.exists)
       format.format(new Date(file.lastModified()))
     else {
-      format.setTimeZone(TimeZone.getTimeZone("UTC"))
-      format.format(new Date(0L))
+      epoch
     }
   }
+
+  private def epoch: String = {
+    val sdf = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss Z")
+    sdf.setTimeZone(TimeZone.getTimeZone("UTC"))
+    sdf.format(new Date(0L))
+  }
+
+  def newFileDiff(text: Seq[String], file: File): String = {
+    val diff = StringBuilder.newBuilder
+    val originalInfo = s"${file.getAbsolutePath} $epoch"
+    val revisedInfo = s"${file.getAbsolutePath} ${fileModificationTimeOrEpoch(file)}"
+
+    diff ++= s"--- $originalInfo$lineSep"
```

Comment:
I tried to generate the diff using `DIffUtils`, but it has the following code in it
```
int origStart = curDelta.getOriginal().getPosition() + 1 - contextSize;
if (origStart < 1) {
	origStart = 1;
}

int revStart = curDelta.getRevised().getPosition() + 1 - contextSize;
if (revStart < 1) {
	revStart = 1;
}
```
Which means `-0,0` or `+0,0` never appears in the diff hunk identifier. I may be missing something, but I though it was required to create/delete file.